frontend/lib/resolution/resolution-queries.cpp

```diff
@@ -2169,14 +2185,55 @@ scopeResolveFunctionQuery(Context* context, ID id) {
     checkForParenlessMethodFieldRedefinition(context, fn, visitor);
 
     sig = visitor.typedSignature;
+
+    if (!visitor.outerVars->isEmpty()) {
+      std::swap(outerVars, visitor.outerVars);
+    }
   }
 
+  QUERY_STORE_RESULT(computeOuterVariablesQuery,
+                     context,
+                     std::move(outerVars),
+                     id);
+
   result = toOwned(new ResolvedFunction(sig, fn->returnIntent(),
                                         std::move(resolutionById),
                                         PoiInfo(),
                                         QualifiedType()));
+  return result;
+}
 
-  return QUERY_END(result);
+static const owned<ResolvedFunction>&
+scopeResolveFunctionQuery(Context* context, ID id) {
+  QUERY_BEGIN(scopeResolveFunctionQuery, context, id);
+  const bool computeOuterVars = true;
+  owned<ResolvedFunction> ret;
+  ret = scopeResolveFunctionQueryBody(context, id, computeOuterVars);
+  return QUERY_END(ret);
+}
+
+const OuterVariables* computeOuterVariables(Context* context, ID id) {
+
+  // For now, preemptively return 'nullptr' if 'id' is not a function.
+  if (!parsing::idIsNestedFunction(context, id)) return nullptr;
+
+  if (!context->hasCurrentResultForQuery(computeOuterVariablesQuery, { id })) {
+    if (!context->isQueryRunning(scopeResolveFunctionQuery, { id })) {
+
+      // The 'computeOuterVariablesQuery' is set as a side effect of
+      // performing scope resolution, since both require a traversal.
+      std::ignore = scopeResolveFunction(context, id);
```

Comment:
A major question I have about this is the following: I believe long term, "scope resolution" will not exist, and be replaced by resolution. At least, this was my understanding of long trm Dyno goals. (under my understanding, scope resolution only exists to get more of the production compiler using Dyno sooner). If that were the case, will this still be workable? Will you need to run resolution to get out variables, and will you need outer variables to run resolution?

Reply:
We'll just cannibalize the scope-resolution code and use it to collect outer variables.